src/MakingSense.AspNetCore.Authentication.SimpleToken/SimpleTokenAuthenticationHandler.cs

```diff
@@ -75,33 +84,39 @@ public static string ExtractToken(HttpRequest request)
 		/// <returns></returns>
 		protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 		{
-			var token = ExtractToken(Request);
-
-			// If no token found, no further work possible
-			if (string.IsNullOrEmpty(token))
+			try
 			{
-				return AuthenticateResult.Skip();
-			}
+				var token = ExtractToken(Request);
 
-			var validationParameters = Options.TokenValidationParameters.Clone();
+				// If no token found, no further work possible
+				if (string.IsNullOrEmpty(token))
+				{
+					return AuthenticateResult.NoResult();
+				}
 
-			SecurityToken validatedToken;
-			var validators = Options.SecurityTokenValidatorsFactory();
-			foreach (var validator in validators)
-			{
-				if (validator.CanReadToken(token))
+				var validationParameters = Options.TokenValidationParameters.Clone();
+
+				var validators = Options.SecurityTokenValidatorsFactory();
+				foreach (var validator in validators)
 				{
-					var principal = validator.ValidateToken(token, validationParameters, out validatedToken);
-					var ticket = new AuthenticationTicket(principal, new AuthenticationProperties(), Options.AuthenticationScheme);
-					return AuthenticateResult.Success(ticket);
+					if (validator.CanReadToken(token))
+					{
+						var principal = validator.ValidateToken(token, validationParameters, out SecurityToken validatedToken);
+						var ticket = new AuthenticationTicket(principal, Scheme.Name);
+						return AuthenticateResult.Success(ticket);
+					}
 				}
-			}
 
-			// Ugly patch to make this method should to be async in order to allow result caching by caller
-			await DoneTask;
+				// Ugly patch to make this method should to be async in order to allow result caching by caller
+				await DoneTask;
```

Comment:
If I am not wrong, now there is a predefined property available in the framework.

Reply:
Thanks, now I know that property is available to use.